File: app/src/main/java/paul/barthuel/go4lunch/ui/chat/ChatDiffCallBack.java
package paul.barthuel.go4lunch.ui.chat;

import androidx.annotation.NonNull;

public class ChatDiffCallBack extends androidx.recyclerview.widget.DiffUtil.ItemCallback<UiMessage> {

@Override
public boolean areItemsTheSame(@NonNull UiMessage oldItem, @NonNull UiMessage newItem) {
        return oldItem.getMessage().equals(newItem.getMessage());
        }

@Override
public boolean areContentsTheSame(@NonNull UiMessage oldItem, @NonNull UiMessage newItem) {
        return oldItem.getMessage().equals(newItem.getMessage())
                && oldItem.getDate().equals(newItem.getDate())
                && oldItem.getSenderName().equals(newItem.getSenderName());
        }
}

File: app/src/main/java/paul/barthuel/go4lunch/ui/chat/ChatDiffCallBack.kt
package paul.barthuel.go4lunch.ui.chat

import androidx.recyclerview.widget.DiffUtil

class ChatDiffCallBack : DiffUtil.ItemCallback<UiMessage>() {
    override fun areItemsTheSame(oldItem: UiMessage, newItem: UiMessage): Boolean {
        return oldItem.message == newItem.message
    }

    override fun areContentsTheSame(oldItem: UiMessage, newItem: UiMessage): Boolean {
        return oldItem.message == newItem.message && oldItem.date == newItem.date && oldItem.senderName == newItem.senderName
    }
}